src/dbnode/persist/fs/files.go

```diff
@@ -1012,13 +1012,30 @@ func IndexSnapshotFiles(filePathPrefix string, namespace ident.ID) (FileSetFiles
 
 // FileSetAt returns a FileSetFile for the given namespace/shard/blockStart/volume combination if it exists.
 func FileSetAt(filePathPrefix string, namespace ident.ID, shard uint32, blockStart time.Time, volume int) (FileSetFile, bool, error) {
+	pattern := filesetFileForTimeAndIndex(blockStart, volume, anyLowerCaseCharsPattern)
+	// If this is the initial volume, then we need to check if files were written with the legacy file naming (i.e.
+	// without the volume index) so that we can properly locate the fileset.
+	if volume == 0 {
+		dir := ShardDataDirPath(filePathPrefix, namespace, shard)
+		isLegacy, err := isFirstVolumeLegacy(dir, blockStart, checkpointFileSuffix)
```

Comment:
nit: Maybe we should be more specific about Legacy and use a different name here. I can imagine that the meaning of Legacy is going to become confusing over time. Something like w/ a specific version maybe? Might be a good to refactor this out and have the code explicitly declare which version we are handling?

Reply:
I think the method call `isFirstVolumeLegacy` (and the comment on the method call) provides the necessary context to indicate that we're referring to legacy file volumes. Additionally, we use this method to perform similar checks in the codebase. I'm inclined to leave as-is for now and do the refactor when future issues surface.